crates/tx/src/types.rs

```diff
@@ -1048,6 +1092,32 @@ impl Tx {
         }
     }
 
+    /// Add new default commitments to the transaction. Returns false if the
+    /// commitment is already contained in the set
+    #[cfg(any(test, feature = "testing"))]
+    pub fn push_default_inner_tx(&mut self) -> bool {
+        self.header.batch.insert(TxCommitments::default())
+    }
+
+    /// Add a new inner tx to the transaction. Returns `false` if the
+    /// commitments already existed in the collection. This function expects a
+    /// transaction carrying a single inner tx as input
+    pub fn add_inner_tx(&mut self, other: Tx, cmt: TxCommitments) -> bool {
```

Comment:
When the same tx code or data is added, they are not inserted but the commitment still points to the sections, and the sections don't exist.
I had an error when multiple ibc txs were added, i.e. batching with the same tx code.
```
"Mempool validation failed: Wrapper transaction code didn't pass the allowlist checks Tx is not allowed in allowlist parameter"
```

Reply:
You're right, I've pushed a new commit that removes that broken optimization for now. We can bring it back when we'll rework the tx builder

Reply:
Thank you! It has resolved the issue.